Makefile

```diff
@@ -1,37 +1,66 @@
+# Image URL to use all building/pushing image targets
+IMG ?= fuseml-core:latest
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
 
 GO_LDFLAGS:=-ldflags '-s -w'
-GO_GCFLAGS=
 
-all: gen-core build
+all: fuseml
+
+# Run tests
+test: generate lint
+	go test ./... -coverprofile cover.out
+
+# Build FuseML binaries
+fuseml: generate lint
+	go build ${GO_LDFLAGS} -o bin/fuseml_core ./cmd/fuseml_core
+	go build ${GO_LDFLAGS} -o bin/fuseml_core-cli ./cmd/fuseml_core-cli
 
-build: tidy lint
-	go build ${GO_GCFLAGS} -o bin/fuseml_core ${GO_LDFLAGS} ./cmd/fuseml_core
-	go build ${GO_GCFLAGS} -o bin/fuseml_core-cli ${GO_LDFLAGS} ./cmd/fuseml_core-cli
+# Run fuseml_core
+run-fuseml_core: generate lint
+	go run ./cmd/fuseml_core
 
-test:
-	ginkgo ./gen ./pkg
+# Run fuseml_core-cli
+run-fuseml_core-cli: generate lint
+	go run ./cmd/fuseml_core-cli
 
-gen-core:
+# Generate code
+generate:
```

Comment:
I am not so fond of having this instruction there. It's neat and convenient, but it's so easy that we'll most likely use and abuse this. Generating a new API means breaking existing APi (if I am not mistaken).
I can foresee this becoming a problem in the long run when we'll have at least ONE api user (and because this project is public, I expect that it will arrive soon, at first installation ;) )

So I would rather this be documented than a CLI function. But maybe I am too careful here.

Reply:
Sorry but I did not get it. Could you elaborate?

If there is no change on `design` the API generated code won't change. However if you change `design` it will change and I believe having generate here encourages finding a breaking change early.

Reply:
Why do we check out the generated code in git, then?

I am fine either way, I just want to understand :)